File: frontend/lib/domain/entities/course_discipline.dart
import 'discipline.dart';

class CourseDiscipline {
  const CourseDiscipline({
    required this.id,
    required this.courseId,
    required this.disciplineId,
    this.discipline,
  });

  final String id;
  final String courseId;
  final String disciplineId;
  final Discipline? discipline;
}

File: backend/src/modules/academic/domain/course-discipline.ts
export type CourseDiscipline = {
  id: string;
  teacherId: string;
  courseId: string;
  disciplineId: string;
  createdAt: Date;
  deletedAt: Date | null;
};

/** Vínculo enriquecido com dados da disciplina, usado em listagens por curso. */
export type CourseDisciplineDetail = CourseDiscipline & {
  discipline: {
    id: string;
    name: string;
    description: string | null;
  };
};
